server/repository/src/db_diesel/sync_file_reference.rs

```diff
@@ -0,0 +1,153 @@
+use super::{
+    sync_file_reference_row::{
+        sync_file_reference, sync_file_reference::dsl as sync_file_reference_dsl,
+    },
+    StorageConnection,
+};
+
+use crate::{
+    diesel_macros::{apply_equal_filter, apply_sort_no_case},
+    SyncFileReferenceRow,
+};
+
+use crate::{repository_error::RepositoryError, DBType, EqualFilter, Pagination, Sort};
+use diesel::prelude::*;
+
+#[derive(PartialEq, Debug, Clone)]
+pub struct SyncFileReference {
+    pub sync_file_reference_row: SyncFileReferenceRow,
+}
+
+#[derive(Clone, PartialEq, Debug, Default)]
+pub struct SyncFileReferenceFilter {
```

Comment:
Might want file_name as some stage,  but no need to add that yet.

Reply:
thanks - I suspect I've already added 'future functionality' i.e. stuff that I'm not using yet. there's plenty of potential here!